fs/vfs/fs_eventfd.c

```
@@ -0,0 +1,600 @@
+          fds->priv   = &dev->fds[i];
+          break;
+        }
+    }
+
+  if (i >= CONFIG_EVENT_FD_NPOLLWAITERS)
+    {
+      fds->priv = NULL;
+      ret       = -EBUSY;
+      goto errout;
+    }
+
+  /* Notify the POLLOUT event if the pipe is not full, but only if
+   * there is readers.
+   */
+
+  eventset = 0;
+  if (dev->counter < (eventfd_t)-1)
+    {
+      eventset |= POLLOUT;
+    }
+
+  /* Notify the POLLIN event if the pipe is not empty */
+
+  if (dev->counter > 0)
+    {
+      eventset |= POLLIN;
+    }
+
+  if (eventset)
+    {
+      eventfd_pollnotify(dev, eventset);
+    }
+
+errout:
+  nxsem_post(&dev->exclsem);
+  return ret;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int eventfd(unsigned int count, int flags)
+{
+  int ret;
+  int new_fd;
+  unsigned int new_minor;
+  FAR struct eventfd_priv_s *new_dev;
+
+  /* devpath: EVENT_FD_VFS_PATH + /efd (4) + %d (3) + null char (1) */
+
+  char devpath[sizeof(CONFIG_EVENT_FD_VFS_PATH) + 4 + 3 + 1];
+
+  /* Allocate instance data for this driver */
+
+  new_dev = eventfd_allocdev();
+  if (new_dev == NULL)
+    {
+      /* Failed to allocate new device */
+
+      ret = ENOMEM;
+      goto exit_set_errno;
+    }
+
+  new_dev->counter = count;
+  new_dev->mode_semaphore = !!(flags & EFD_SEMAPHORE);
+
+  /* Request a unique minor device number */
+
+  new_minor = eventfd_get_unique_minor();
+
+  if (new_minor < 0)
+    {
+      ferr("Cannot get minor\n");
+      ret = EMFILE;
+      goto exit_free_new_dev;
+    }
+
+  new_dev->minor = new_minor;
+
+  /* Get device path */
+
+  sprintf(devpath, CONFIG_EVENT_FD_VFS_PATH "/efd%d", new_minor);
+
+  /* Register the driver */
+
+  ret = register_driver(devpath, &g_eventfd_fops, 0666, new_dev);
+  if (ret < 0)
+    {
+      ferr("Failed to register new device %s: %d\n", devpath, ret);
+      ret = ENOMEM;
```

why eat the error code returned from register_driver? it's better to return the value as it.